src/operators.jl

```diff
@@ -858,3 +858,191 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where U <: Integer) where T
 
     return new_vertex_ids
 end
+
+"""
+    linegraph(g)
+
+returns only the line graph of a given graph if returnmap is false
+returns line graph and a dictionary mapping new vertices in line graph to
+old edges in graph if returnmap is true
+"""
+function linegraph(g::AbstractGraph, returnmap::Bool = false)
+    lg = is_directed(g) ? SimpleDiGraph(ne(g)) : SimpleGraph(ne(g))
+    graphE_lineV = Dict(e => i for (i, e) in enumerate(edges(g)))
+    earr = collect(edges(g))
+    for i in 1:(ne(g)-1)
+        for j in i+1:ne(g)
```

Comment:
I think if we do it like that, we will miss self-loops.

Reply:
https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.generators.line.line_graph.html

Reply:
I don't think self loops should be included

Reply:
I will have to do some resarch on undirected graphs, but at least for directed graphs, I think they should be included as then you get some beautiful stuff like: https://en.wikipedia.org/wiki/De_Bruijn_graph